MobiusCore/Source/AsyncDispatchQueueConnectable.swift

```diff
@@ -47,21 +38,14 @@ final class AsyncDispatchQueueConnectable<Input, Output>: Connectable {
         self.init(AnyConnectable(underlyingConnectable), acceptQueue: acceptQueue)
     }
 
-    func connect(_ consumer: @escaping (Output) -> Void) -> Connection<Input> {
-        let disposalStatus = Synchronized(value: DisposalStatus.notDisposed)
+    func connect(_ consumer: @escaping Consumer<Output>) -> Connection<Input> {
+        let protectedConsumer = Synchronized<Optional<Consumer<Output>>>(value: consumer)
```

Comment:
Synchronizing on the consumer allows for both tracking the disposal status and clearing the reference when it is no longer usable.

Reply:
Maybe just bikeshedding but this might make for a good implementation comment, especially considering how difficult it can be to read the nested parameterized types.